src/fides/api/service/connectors/dynamic_erasure_email_connector.py

```diff
@@ -0,0 +1,258 @@
+        # We'll batch identities for each email address
+        # so we'll only send 1 email to each email address
+        batched_identities: Dict[str, List[str]] = {}
+
+        for privacy_request in privacy_requests:
+            email = self.get_email_address_from_custom_request_fields(
+                connector,
+                graph_dataset,
+                privacy_request,
+                collection_address,
+                field_name,
+            )
+
+            user_identities: Dict[str, Any] = privacy_request.get_cached_identity_data()
+            filtered_user_identities: Dict[str, Any] = (
+                filter_user_identities_for_connector(self.config, user_identities)
+            )
+
+            # If we couldn't get the email or the user identities, we skip the privacy request
+            if not email or not filtered_user_identities:
+                skipped_privacy_requests.append(privacy_request.id)
+                self.add_skipped_log(db, privacy_request)
+                continue
+
+            if email not in batched_identities:
+                batched_identities[email] = []
+
+            batched_identities[email].extend(filtered_user_identities.values())
+
+        if not batched_identities:
+            logger.info(
+                "Skipping erasure email send for connector: '{}'. "
+                "No corresponding user identities or email addresses found for pending privacy requests.",
+                self.configuration.name,
+            )
+            return
+
+        logger.info(
+            "Sending batched erasure email for connector {}...",
+            self.configuration.name,
+        )
+
+        for email_address, identities in batched_identities.items():
+            try:
+                send_single_erasure_email(
+                    db=db,
+                    subject_email=email_address,
+                    subject_name=self.config.third_party_vendor_name,
+                    batch_identities=identities,
+                    test_mode=False,
+                )
+            except MessageDispatchException as exc:
+                logger.info("Erasure email failed with exception {}", exc)
+                raise
+
+        # create an audit event for each privacy request ID
+        for privacy_request in privacy_requests:
+            if privacy_request.id not in skipped_privacy_requests:
+                ExecutionLog.create(
+                    db=db,
+                    data={
+                        "connection_key": self.configuration.key,
+                        "dataset_name": self.configuration.name,
+                        "collection_name": self.configuration.name,
+                        "privacy_request_id": privacy_request.id,
+                        "action_type": ActionType.erasure,
+                        "status": ExecutionLogStatus.complete,
+                        "message": f"Erasure email instructions dispatched for '{self.configuration.name}'",
```

Comment:
These logs help inform what shows up here:

<img width="1019" alt="Screenshot 2024-08-25 at 2 18 49 PM" src="https://github.com/user-attachments/assets/4b6da47a-bbdb-462e-a29f-0157dc31a083">

Reply:
So that's copied from the existing email connector , which uses the configuration name. Do you still think we should change it? maybe we can use the name , and fallback to the key if the name doesn't exist?

Reply:
yes fallback to key if name doesn't exist is a good idea

Reply:
just pushed the change